### PR TITLE
chore: Emit deprecation messages

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
This pull request emits deprecation warnings to deprecated parts of the library.

This was not done for `MessageEmbed#type` as that was just a property (though it could technically be made into a getter?) and the application events that don't emit at all.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by vladfrangu — APPROVED** [review]

File: src/util/Util.js [hunk 1]
```diff
@@ -578,6 +580,15 @@ class Util extends null {
    * @deprecated Use {@link BaseMessageOptions#allowedMentions} instead.
    */
   static removeMentions(str) {
+    if (!deprecationEmittedForRemoveMentions) {
+      process.emitWarning(
+        'This method is deprecated. Utilise allowed mentions when sending a message instead.',
```
Comment (ImRodry) [inline src/util/Util.js:585]: Why doesn't this follow the same pattern as the others?
```suggestion
        'This method is deprecated. Use MessageOptions#allowedMentions when sending a message instead.',
```

**Review by ImRodry — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]
I'm guessing that the typings already have `@deprecated`, LGTM.

File: src/structures/Guild.js [hunk 2]
```diff
@@ -1195,6 +1198,15 @@ class Guild extends AnonymousGuild {
    *   .catch(console.error);
    */
   setChannelPositions(channelPositions) {
+    if (!deprecationEmittedForSetChannelPositions) {
+      process.emitWarning(
+        'This method (Guild#setChannelPositions) is deprecated. Use GuildChannelManager#setPositions instead.',
```
Comment (ImRodry) [inline src/structures/Guild.js:1203]: Not gonna suggest this for all as that'd be too spammy but what do you think of this format?
```suggestion
        'This Guild#setChannelPositions method is deprecated. Use GuildChannelManager#setPositions instead.',
```

Comment (almeidx) [reply src/structures/Guild.js:1203]: > The Guild#setChannelPositions method is deprecated

Comment (ImRodry) [reply src/structures/Guild.js:1203]: Fixed

**Comment by suneettipirneni** [comment]
Another message can be added as a result of #6889

File: src/structures/Guild.js (hunk 2 above)
Comment (Jiralite) [reply src/structures/Guild.js:1203]: Sure, I can do that!

**Comment by advaith1** [comment]
can permission functions warn on usage of the USE_*_THREADS permissions?

**Comment by Jiralite** [comment]
> can permission functions warn on usage of the USE_*_THREADS permissions?

I can for sure check them in `Permissions#any`, `Permissions#has` & `Permissions#missing` but I'm not sure if I can for the rest of the methods. They're implemented from the parent class of which conflicting bits may be a possibility from other extended classes... but it's a start?

**Pull request merged**